addon/components/select-2.js

```diff
@@ -463,6 +498,22 @@ var Select2Component = Ember.Component.extend({
         // treat data as a single object
         value = get(data, optionValuePath);
       }
+    } else if (this.get('tags') && e) {
+      if (e.added) {
+        var record;
+        if (e.added instanceof DS.Model) {
+          record = e.added;
+        } else {
+          record = this.createRecordFromTag(e.added);
+        }
+        this.get('value').pushObject(record);
+      } else if (e.removed) {
+        // TODO: Is this always the preferred behaviour? Maybe make this configurable...
+        this.get('value').removeObject(e.removed);
+        if (e.removed.get('isNew')) {
+          e.removed.unloadRecord();
+        }
```

Comment:
I think that this is the correct default behaviour, yes. If there is a need for changing this behaviour, we could easily add a `deleteRemovedTags` option (that would default to `true` for backwards compatibility). But that would mean yet another test case for now, I'd say we leave it that way.